src/PowerShellEditorServices/Language/FindDotSourcedVisitor.cs

```diff
@@ -32,15 +39,44 @@ public FindDotSourcedVisitor()
         /// or a decision to continue if it wasn't found</returns>
         public override AstVisitAction VisitCommand(CommandAst commandAst)
         {
-            if (commandAst.InvocationOperator.Equals(TokenKind.Dot) &&
-                commandAst.CommandElements[0] is StringConstantExpressionAst)
+            CommandElementAst commandElementAst = commandAst.CommandElements[0];
+            if (commandAst.InvocationOperator.Equals(TokenKind.Dot))
             {
-                // Strip any quote characters off of the string
-                string fileName = commandAst.CommandElements[0].Extent.Text.Trim('\'', '"');
-                DotSourcedFiles.Add(fileName);
+                if (commandElementAst is StringConstantExpressionAst stringConstantExpressionAst)
+                {
+                    // Strip any quote characters off of the string
+                    DotSourcedFiles.Add(PathUtils.NormalizePathSeparators(stringConstantExpressionAst.Value));
+                }
+                else if (commandElementAst is ExpandableStringExpressionAst expandableStringExpressionAst)
+                {
+                    var path = GetPathFromExpandableStringExpression(expandableStringExpressionAst);
+                    if (path != null)
+                    {
+                        DotSourcedFiles.Add(PathUtils.NormalizePathSeparators(path));
+                    }
+                }
             }
 
             return base.VisitCommand(commandAst);
         }
+
+        private string GetPathFromExpandableStringExpression(ExpandableStringExpressionAst expandableStringExpressionAst)
+        {
+            var path = expandableStringExpressionAst.Value;
+            foreach (var nestedExpression in expandableStringExpressionAst.NestedExpressions)
```

Comment:
What does `NestedExpressions` mean here? Might be worth adding a comment to explain what the foreach loop is doing.

Reply:
I added a comment in my new commit. Let me know if it's clear enough.